### PR TITLE
GItcoin Bounty PR

### PR DESCRIPTION
**Pull request by Pfed-prog**
https://github.com/strips-finance/apy/issues/1

The three points completed:

1. Supply USDC to compound.finance
2. Track cUSDC profit each 10 minutes (by redeem on fork mainnet)
3. Provide the APY - cumulative yearly rate.

### CODE REVIEW

File: test/cUSDC.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    const owner = await generate_Account();
+
+    const cUSDC = new ethers.Contract(Contracts.cUSDC, compAbi["cErc20Delegate"], owner);
+
+    const ethMantissa = 1e18;
+    const blocksPerDay = 6570; // 13.15 seconds per block
+    const daysPerYear = 365;
+    const supplyRatePerBlock = await cUSDC.supplyRatePerBlock();
+    
+    const supplyApy = (((Math.pow((supplyRatePerBlock / ethMantissa * blocksPerDay) + 1, daysPerYear))) - 1) * 100;
```
Comment (kshilov) [inline test/cUSDC.ts:113]: Is that cumulative in real contract? When you receive profit you don't re-stake it. How could it be cummulaitve?

File: test/cUSDC.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    console.log("New cUSDC balance:",ethers.utils.formatUnits(await cUSDC.balanceOf(owner.address), 8));
+
+  });
+  it("2. Track the price of cUSDC", async () => {
+    
+    const owner = await generate_Account();
+
+    const cUSDC = new ethers.Contract(Contracts.cUSDC, compAbi["cErc20Delegate"], owner);
+
+    checkCurrentBlock()
```
Comment (kshilov) [inline test/cUSDC.ts:95]: What is that for? It will not be per block anyway.

File: test/cUSDC.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    let tx = await cUSDC.redeem(await cUSDC.balanceOf(owner.address));
```
Comment (kshilov) [inline test/cUSDC.ts:96]: We don't nee to redeem each block - precision will not allow us to calc it carefully.

File: test/cUSDC.ts [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+    const cUSDC = new ethers.Contract(Contracts.cUSDC, compAbi["cErc20Delegate"], owner);
+
+    const ethMantissa = 1e18;
+    const blocksPerDay = 6570; // 13.15 seconds per block
+    const daysPerYear = 365;
+    const supplyRatePerBlock = await cUSDC.supplyRatePerBlock();
+    
+    const supplyApy = (((Math.pow((supplyRatePerBlock / ethMantissa * blocksPerDay) + 1, daysPerYear))) - 1) * 100;
+    console.log(`Supply APY for ETH ${supplyApy} %`);
```
Comment (kshilov) [inline test/cUSDC.ts:114]: Didn't find COMP calculation. We can't calc APY without including COMP rewards

**Review by kshilov — CHANGES_REQUESTED** [review]

**Comment by Pfed-prog** [comment]
Gotta learn how to write proper issues. Your previous code was atrocious.

**Comment by kshilov** [comment]
The issue is not implemented as required.

**Pull request closed without merging**